controllers/host/host_controller.go

```diff
@@ -1875,22 +1875,29 @@ func (r *HostReconciler) ReconcileResource(
 
 	if !instance.DeletionTimestamp.IsZero() {
 		if utils.ContainsString(instance.ObjectMeta.Finalizers, HostFinalizerName) {
+
+			// Remove the finalizer so we don't try to do this delete action again.
+			// Defer the removal of the finalizer until the end of the function
+			defer func() {
```

Comment:
1. By adding the defer function, it will always triggered in the scope of ReconcileResource, which is wrong, it should be only executed when !instance.DeletionTimestamp.IsZero(), you'll need to create a new method to wrap the section in the if block.
2. You don't need to change the location of the section as it is expected to be always triggered with defer
3. Prefer to have a named function for the defer function for ease to unit test

Reply:
The defer statement executes only when !instance.DeletionTimestamp.IsZero() is true
Also it will be executed in the scope of the if block because https://github.com/Wind-River/cloud-platform-deployment-manager/blob/master/controllers/host/host_controller.go#L1904 is returning inside the block